src/lib/operators/join_nested_loop.cpp

```diff
@@ -154,7 +155,7 @@ void JoinNestedLoop::_perform_join() {
 
           // clang-format off
           if constexpr (neither_is_string_column || both_are_string_columns) {
-            auto iterable_left = create_iterable_from_column<LeftType>(typed_left_column);
+            auto iterable_left = erase_type_from_iterable(create_iterable_from_column<LeftType>(typed_left_column));
```

Comment:
General discussion: What does this change mean for tests? How can we make sure that we don't add cases where tests behave differently in debug/release?

Reply:
As long as the `AnyColumnIterable` works as intended (there is a test now), tests will behave in the exact same manner since the same code is called (+ the small wrapper in the debug build).